packages/backend/error/src/api-error.ts

```diff
@@ -43,6 +44,40 @@ export default class APIError<C> extends Error {
     this.param = param;
   }
 
+  /**
+   * Sets the ErrorType according to the given HTTP response statusCode
+   *
+   * @param statusCode - The statusCode of the HTTP response
+   */
+  public static httpStatusCodeToErrorType(statusCode: number): ErrorType {
+    switch (statusCode) {
+      case 400:
+        return ErrorType.InvalidData;
+
+      case 401:
+        return ErrorType.AuthenticationError;
+
+      case 403:
+        return ErrorType.ForbiddenError;
+
+      case 404:
+        return ErrorType.NotFoundError;
+
+      case 422:
+        return ErrorType.ApiConectionError;
+
+      case 429:
+        return ErrorType.TooManyRequests;
+
+      case 502:
+        return ErrorType.ApiConectionError;
+
+      case 500:
+      default:
+        return ErrorType.UnknownError;
```

Comment:
If it's a 500, we know it's a server error, so it would be more accurate to say this. 
```suggestion
      case 500:
        return ErrorType.ServerError;
      default:
        return ErrorType.UnknownError;
```

Would need to create `ErrorType.ServerError` too.

Reply:
Good shout

Reply:
However, this would still return the following when `errorTypeToHttpStatusCode()` is called,

```typescript
case ErrorType.UnknownError:
      default:
        return 500;
```